### ISSUE TITLE
Make ReactiveFragment call the appropriate VM lifecycle methods

### ISSUE THREAD

**Issue by heytherewill** [toggl/mobileapp#4866]
Depends on #4863 

We need to define what methods of Fragment map to the ViewModel lifecycle methods we decide to stick to when working on #4863. Then we need to make `ReactiveFragment` call those methods.

**Comment by daividssilverio** [comment]
About the `ReactiveDialogFragment`, the lifecycle is is also alright.
It's almost identical to the Activity, except by the `AttachView`/`DetachView` calls, which on the DialogFragment happen on `OnViewCreated` and `OnDestroyView` respectively.

**Comment by heytherewill** [comment]
Keeping this issue open and removing Daivid's assignment because we still need to deal with some corner cases, but that's for later.

**Comment by daividssilverio** [comment]
I've double checked the lifecycles of our `TabFragments` and `DialogFragments`, alongside with `Activities` as well.

Found nothing critical related to the lifecycle itself. I say we ready to close this issue.

On the other hand, I found some leakage happening in the `Activities` and the cursors opened in the calendar service. I'll open other issues for that, as they were there before and not introduced by mvexit.

Feel free to close this issue at your discretion.
